docker/docker-compose.1804.53.yaml

```diff
@@ -0,0 +1,48 @@
+version: "3"
+
+services:
+
+  runtime-setup:
+    image: swift-nio:18.04-5.3
+    build:
+      args:
+        base_image: "swiftlang/swift:nightly-5.3-bionic"
+
+  unit-tests:
+    image: swift-nio:18.04-5.3
+
+  integration-tests:
+    image: swift-nio:18.04-5.3
+
+  test:
+    image: swift-nio:18.04-5.3
+    environment:
+      - MAX_ALLOCS_ALLOWED_1000_reqs_1_conn=30540
+      - MAX_ALLOCS_ALLOWED_1_reqs_1000_conn=486500 #5.3 improvement 491050
+      - MAX_ALLOCS_ALLOWED_ping_pong_1000_reqs_1_conn=4440
+      - MAX_ALLOCS_ALLOWED_bytebuffer_lots_of_rw=2100
+      - MAX_ALLOCS_ALLOWED_future_lots_of_callbacks=75010
+      - MAX_ALLOCS_ALLOWED_creating_10000_headers=100
+      - MAX_ALLOCS_ALLOWED_scheduling_10000_executions=20150
+      - MAX_ALLOCS_ALLOWED_modifying_1000_circular_buffer_elements=50
+      - MAX_ALLOCS_ALLOWED_encode_1000_ws_frames_holding_buffer=1010
+      - MAX_ALLOCS_ALLOWED_encode_1000_ws_frames_holding_buffer_with_space=1010
+      - MAX_ALLOCS_ALLOWED_encode_1000_ws_frames_new_buffer=4010
+      - MAX_ALLOCS_ALLOWED_encode_1000_ws_frames_new_buffer_with_space=4010
+      - MAX_ALLOCS_ALLOWED_decode_1000_ws_frames=1000
+      - MAX_ALLOCS_ALLOWED_modifying_byte_buffer_view=2010 
+      - MAX_ALLOCS_ALLOWED_schedule_10000_tasks=90050
+      - MAX_ALLOCS_ALLOWED_read_10000_chunks_from_file=200500 #5.3 improvement 210050
+      # - SANITIZER_ARG=--sanitize=thread (filed a bug with the thread sanitizer)
```

Comment:
that's rdar://58953747 . Could you change this docker image to run on Ubuntu 16.04? I think that'll make it work.

Reply:
Don't we want an even spread?

Reply:
@Davidde94 Good idea. We could decide that only the latest two (5.2 as latest official; 5.3 as upcoming) get TSan and the older two (5.0 & 5.1) don't get TSan but get 18.04.

So:
- 5.0: 18.04, no TSan
- 5.1: 18.04, no TSan
- 5.2: 16.04, TSan
- 5.3: 16.04, TSan

@tomerd / @Davidde94 wdyt?

Reply:
That sounds good to me :)